src/mango/src/mango_idx.erl

```diff
@@ -93,17 +109,31 @@ recover(Db) ->
     end, DDocs)}.
 
 
-for_sort(Indexes, Opts) ->
-    % If a sort was specified we have to find an index that
-    % can satisfy the request.
+has_use_index(Opts) ->
+    case lists:keyfind(sort, 1, Opts) of
+        {use_index, _} ->
```

Comment:
This check will never return true as sort is not equal to use_index.

Reply:
Which means we're likely missing a test for this since travis returned green.

Reply:
There is a test but this situation is handled elsewhere. I'll remove `has_use_index` and its usage.